src/main/kotlin/com/mdrobnak/lalrpop/intentions/AddExplicitTypeIntention.kt

```diff
@@ -0,0 +1,64 @@
+package com.mdrobnak.lalrpop.intentions
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import com.mdrobnak.lalrpop.LpLanguage
+import com.mdrobnak.lalrpop.psi.LpAlternative
+import com.mdrobnak.lalrpop.psi.LpElementFactory
+import com.mdrobnak.lalrpop.psi.LpNonterminal
+import com.mdrobnak.lalrpop.psi.getContextAndResolveType
+
+class AddExplicitTypeIntention : IntentionAction {
+    override fun startInWriteAction(): Boolean = true
+
+    override fun getText(): String = "Add explicit type"
+
+    override fun getFamilyName(): String = "Add explicit type"
+
+    override fun isAvailable(project: Project, editor: Editor?, file: PsiFile?): Boolean {
+        if (editor == null || file == null || file.language != LpLanguage) return false
+
+        var element = file.findElementAt(editor.caretModel.primaryCaret.offset)
+        while (element != null) {
+            if (element is LpNonterminal) return true
+            element = element.parent
+        }
+        return false
+    }
```

Comment:
> It looks like your intention may be to allow updating an explicit type in case the type is initially incorrect, but would that be better handled in WrongInferredTypeInspection?

Yes, more or less

> We could extract the logic to find and set/replace the explicit type into a quick fix and execute it in the inspection and in this intention.

Ok I randomly happened to bump into `IntentionAndQuickFixAction` that they can both use